docs/slsa.md

```diff
@@ -0,0 +1,338 @@
+# SLSA provenance
+
+BuildKit supports automatic creation of [SLSA Provenance](https://slsa.dev/provenance)
+attestations to record the build process of the image, including details such
+as:
+
+- Build timestamps
+- Build parameters and environment
+- Version control metadata
+- Source code details
+- Materials consumed during the build
+
+All provenance generated by BuildKit is wrapped inside [in-toto attestations](https://github.com/in-toto/attestation)
+in the [SLSA Provenance format](https://slsa.dev/provenance/v0.2).
+
+When the final output format is a container image, provenance is attached
+using the [attestation storage](./attestation-storage.md).
+
+To build an image with attached provenance use the `attest:provenance` option:
+
+```bash
+buildctl build \
+    --frontend=dockerfile.v0 \
+    --local context=. \
+    --local dockerfile=. \
+    --opt attest:provenance=
+```
+
+You can also specify custom parameters for the provenance:
+
+```bash
+buildctl build \
+    --frontend=dockerfile.v0 \
+    --local context=. \
+    --local dockerfile=. \
+    --opt attest:provenance=mode=min,inline-only=true
+```
+
+## Parameters
+
+| Parameter      | Type           | Default          | Description                                                                                                 |
+| -------------- | -------------- | ---------------- | ----------------------------------------------------------------------------------------------------------- |
+| `mode`         | `min`,`max`    | `max`            | Configures the amount of provenance to be generated. See [mode](#mode)                                      |
+| `builder-id`   | String         |                  | Explicitly set SLSA [`builder.id`](https://slsa.dev/provenance/v0.2#builder.id) field                       |
+| `reproducible` | `true`,`false` | `false`          | Explicitly set SLSA [`metadata.reproducible`](https://slsa.dev/provenance/v0.2#metadata.reproducible) field |
```

Comment:
Setting `SOURCE_DATE_EPOCH` might influence this field as well as build timestamps?